llvm/include/llvm/TableGen/CodeGenHelpers.h

```diff
@@ -46,6 +46,24 @@ class IfDefEmitter {
   bool LateUndef;
 };
 
+// Simple RAII helper for emitting #if guard. It emits:
+// <IfGuard> <Condition>
+// #endif // <Condition>
+class IfGuardEmitter {
+public:
+  IfGuardEmitter(raw_ostream &OS, StringRef Condition,
+                 StringRef IfGuard = "#if")
```

Comment:
Why is `#if` a parameter?

Reply:
If it helps, I can attempt to migrate one such case as well, so it's covered in this PR.

Reply:
Also, `#ifdef X` could be rewritten as `#if defined X`, which is generally preferred as it allows adding more `&&`s and `||`s in the future

Reply:
Agreed, maybe we can keep it `#if` always and generalize if need be in future?

Reply:
Done.